lib/internal/modules/esm/loader.js

```diff
@@ -148,29 +148,29 @@ function nextHookFactory(chain, meta, { validateArgs, validateOutput }) {
   }
 
   return ObjectDefineProperty(
-    async (...args) => {
+    async (arg0, context) => {
       // Update only when hook is invoked to avoid fingering the wrong filePath
       meta.hookErrIdentifier = `${hookFilePath} '${hookName}'`;
 
-      validateArgs(`${meta.hookErrIdentifier} hook's ${nextHookName}()`, args);
+      validateArgs(`${meta.hookErrIdentifier} hook's ${nextHookName}()`, [arg0, context]);
 
       const outputErrIdentifier = `${chain[generatedHookIndex].url} '${hookName}' hook's ${nextHookName}()`;
 
       // Set when next<HookName> is actually called, not just generated.
       if (generatedHookIndex === 0) { meta.chainFinished = true; }
 
-      // `context` is an optional argument that only needs to be passed when changed
-      switch (args.length) {
-        case 1: // It was omitted, so supply the cached value
-          ArrayPrototypePush(args, meta.context);
-          break;
-        case 2: // Overrides were supplied, so update cached value
-          ObjectAssign(meta.context, args[1]);
-          break;
+      const argsToApply = [arg0];
+
+      if (context) { // `context` has already been validated, so no fancy check needed.
+        argsToApply[1] = context;
+        ObjectAssign(meta.context, context);
+      } else {
+        argsToApply[1] = meta.context;
       }
```

Comment:
This could maybe be simplified to:

```js
if (context) {
  ObjectAssign(meta.context, context);
}

argsToApply[1] = meta.context;
```

But that may lead to unintended change(s) in behaviour 🤔